### ISSUE TITLE
[SODA] Form fields with errors not identified

### ISSUE THREAD

**Issue by aaron-collier** [sul-dlss/hungry-hungry-hippo#903]
The form field with an error is not identified in text, or text alternative, or via programmatic association.
There are several issues with the form validation.

- When a field is in error, an error message appears visibly inline, but it is not programmatically associated with a form field. 
- When the error is a required field not entered, the text is a generic "can't be blank"
- Fields that are in error do not have aria-invalid="true" set on them.
- Fields that are required are not indicated as being required visually or programmatically.

Soda recommendation:
```
The form field must be associated with the error text via aria-describedby as outline in the guidance that follows.

The error message text must be descriptive regarding the error and the form field in error. 

When a form field is error, aria-invalid="true" should be added .

Required fields need instructions or labels to visible indicate they are required and the form field must include "required" or aria-required="true"

WebAIM - Usable and Accessible Form Validation and Error Recovery
https://webaim.org/techniques/formvalidation/

aria-required - MDN Web Docs
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-required

RULE : 
Text-based alerts that identify the field in error MUST be provided to users for form validation cues and errors.

HOW TO FIX: 
Fix this issue by doing AT LEAST ONE of the following:
1. Programmatically associate the error message and the field in error - typically using the aria-describedby attribute to reference a unique id attribute value on the error message container. (STRONGLY PREFERRED)
2. Include the field name in the error message.

REFERENCE: 
Deque University: https://dequeuniversity.com/class/forms/form-validation/error-identification
W3C-WAI tutorial: https://www.w3.org/WAI/tutorials/forms/notifications/

BACKGROUND: 
When form fields that have an error are not identified with a text-based error message people who are blind or have low vision may not be able to understand that an error has occurred and what went wrong. In order to provide feedback that can be accessed by all users, error messages must be provided in a text format. Error messages must indicate which input is in error - either by the text of the error message or programmatically - and the nature of the error. This will allow people to recover from mistakes while filling out a form.
```

See issue id 2272 in https://docs.google.com/spreadsheets/d/1x0uBBDe3T8O0HNCs74OdVKQc8tQQNvOV_f-C6IS44IQ/edit?gid=1050712151#gid=1050712151
